exercises/circular-buffer/canonical-data.json

```diff
@@ -0,0 +1,296 @@
+{
+  "#": [
+    "In general, these circular buffers are expected to be stateful,",
+    "and each language will operate on them differently.",
+    "Tests tend to perform a series of operations, some of which expect a certain result.",
+    "As such, this common test suite can only say in abstract terms what should be done.",
+    "",
+    "Tests will contain a number of operations. The operation will be specified in the `operation` key.",
+    "Based on the operation, other keys may be present.",
+    "read: Reading from the buffer should produce the item located at `expected`.",
+    "write: Writing the item located at `item` should succeed.",
+    "readShouldFail: Reading from the buffer should fail.",
+    "writeShouldFail: Writing the item located at `item` should fail.",
+    "overwrite: Write the item located at `item` into the buffer, removing the oldest item if necessary.",
+    "clear: Clear the buffer.",
+    "",
+    "Finally, note that all values are integers.",
+    "If your language contains generics, you may consider allowing buffers to contain other types.",
+    "Tests for that are not included here.",
+    ""
+  ],
+  "cases": [
+    {
+      "description": "reading empty buffer should fail",
+      "capacity": 1,
+      "operations": [
+        {
+          "operation": "readShouldFail"
```

Comment:
This has a little to do with #401, what's the best way to represent an error here. I am having a sufficiently hard time with fitting `"expected": true/false` into all operations that I think I should keep them separate. This is the suggestion of https://github.com/exercism/x-common/issues/401#issuecomment-253519582 - errors are simply encoded differently. I know I didn't use that in any other JSON file I've written, but I managed to make `null` work in all of those, whereas I'm unsure whether integer/null for reads versus true/false for writes gets too confusing. Even integer/false for reads seems a bit odd.

That, and I may be a bit biased by how the [go tests](https://github.com/exercism/xgo/blob/master/exercises/circular-buffer/circular_buffer_test.go#L48) were written - they pretty much use those exact six operations already.